### DISCUSSION TITLE
AIDeck Bootloader not flashing

### DISCUSSION THREAD

**Discussion by vidushij17** [bitcraze/discussions#1468] [Q&A]
Whenever I run 
```
cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w radio://0/80/2M/E7E7E7E7E7
``` 
it only goes up until 0% and stops
```
Reset to bootloader mode ...
nRF51 has: sd-s130 and requires None and upgrade provides None. Current bootloader version is [2024.2.0] but upgrade provides [None]
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcAI:esp, not in the target list
Deck bcAI:gap8, reset to bootloader
| 0% Writing to bcAI:gap8 deck memory
```

Earlier it was going up to 99% but it's not even doing that anymore.

**Comment by evoggy** [comment]
Hi! Do you have any other decks attached, like the Lighthouse deck?

**Reply by vidushij17** [reply]
No, just the AI deck.

**Reply by evoggy** [reply]
Have you ever been able to flash the deck to 100%? Does it work flashing it with JTAG?

**Reply by vidushij17** [reply]
No, the most it has been able to flash is 99%. Still working on acquiring a JTAG.

**Reply by evoggy** [reply]
I would recommend updating all the firmware (including the GAP8 bootloader via JTAG) and try it again.

**Reply by vidushij17** [reply]
I have a ST Link. Would that work for updating the firmware?

**Reply by knmcguire** [reply]
I'm afraid that the ST link willl not be sufficent as that would only work for ST-made microprocessors. We have had success with the Olimex programmer and the Jlink. You can find more information here: https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/jtag-programmer/
